File: training-data/go/atbash_cipher.go
package main

import (
	"fmt"
	"strings"
)

func atbash(s string) string {
	return strings.Map(func(r rune) rune {
		switch {
		case r >= 'a' && r <= 'z':
			return 'z' - (r - 'a')
		case r >= 'A' && r <= 'Z':
			return 'Z' - (r - 'A')
		}
		return r
	}, s)
}

func main() {
	enc := atbash("Hello, World!")
	fmt.Println(enc, "|", atbash(enc))
}

File: training-data/typescript/atbash_cipher.ts
export const atbash = (text: string): string =>
  text.replace(/[a-z]/gi, (ch) => {
    const base = ch <= "Z" ? 65 : 97;
    return String.fromCharCode(base + 25 - (ch.charCodeAt(0) - base));
  });

const enc = atbash("Hello, World!");
console.log(enc, "|", atbash(enc));
